src/ast/element-descriptor.ts

```diff
@@ -29,3 +43,33 @@ export interface ElementDescriptor extends Descriptor {
 
   abstract?: boolean;
 }
+
+/**
+ * The metadata for a single polymer element
+ */
+export class ElementDescriptor implements Descriptor {
```

Comment:
I'd kinda prefer to have two different classes. One for elements, one for behaviors that extend a common class (inheriting most of their properties from same).

That way we could do:


```typescript
if (elementOrBehavior.type === 'element') {
  // type checker knows it's an element here
}
```